### PR TITLE
solution hw-spring-07

### PR DESCRIPTION
**Pull request by RomanHreits**

### CODE REVIEW

**Review by DmGrt — APPROVED** [review]
Seems everything is fine

File: src/main/java/com/cinema/controller/OrderController.java [hunk 1]
```diff
@@ -32,18 +33,23 @@ public OrderController(OrderService orderService, OrderMapper mapper,
...
         return orderService.getOrderHistory(user)
                 .stream()
                 .map(mapper::fromEntityToDto)
                 .collect(Collectors.toList());
     }
+
+    private User getUser(Authentication authentication) {
+        UserDetails principal = (UserDetails) authentication.getPrincipal();
+        return userService.findByEmail(principal.getUsername()).get();
+    }
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/cinema/controller/OrderController.java]: I don't see the point of moving this to separate method but it's up to you

File: src/main/java/com/cinema/lib/FieldsValueMatch.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import javax.validation.Payload;
+
+@Constraint(validatedBy = PasswordValueMatchValidator.class)
+@Target({ ElementType.TYPE })
+@Retention(RetentionPolicy.RUNTIME)
+public @interface FieldsValueMatch {
+ 
+    String message() default "Fields values don't match!";
+    Class<?>[] groups() default {};
+    Class<? extends Payload>[] payload() default {};
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/cinema/lib/FieldsValueMatch.java]:
```suggestion
    String message() default "Fields values don't match!";
    
    Class<?>[] groups() default {};
    
    Class<? extends Payload>[] payload() default {};
```

File: src/main/java/com/cinema/validator/PasswordValueMatchValidator.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        Object fieldValue = new BeanWrapperImpl(value)
+                .getPropertyValue(field);
+        Object fieldMatchValue = new BeanWrapperImpl(value)
+                .getPropertyValue(fieldMatch);
+
+        if (fieldValue != null) {
+            return fieldValue.equals(fieldMatchValue);
+        } else {
+            return fieldMatchValue == null;
+        }
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/cinema/validator/PasswordValueMatchValidator.java]: ternary operator?

File: src/main/java/com/cinema/controller/OrderController.java (hunk 1 above)
Comment (RomanHreits) [reply src/main/java/com/cinema/controller/OrderController.java]: done

File: src/main/java/com/cinema/lib/FieldsValueMatch.java (hunk 2 above)
Comment (RomanHreits) [reply src/main/java/com/cinema/lib/FieldsValueMatch.java]: done

File: src/main/java/com/cinema/validator/PasswordValueMatchValidator.java (hunk 3 above)
Comment (RomanHreits) [reply src/main/java/com/cinema/validator/PasswordValueMatchValidator.java]: done

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**